elide-core/src/main/java/com/yahoo/elide/core/RequestScope.java

```diff
@@ -88,7 +92,25 @@ public RequestScope(JsonApiDocument jsonApiDocument,
 
         newPersistentResources = new LinkedHashSet<>();
         commitTriggers = new LinkedHashSet<>();
-        permissionExecutor = new PermissionExecutor(this);
+        if (permissionExecutorClass == null) {
+            this.permissionExecutor = new ActivePermissionExecutor(this);
+        } else {
+            try {
+                PermissionExecutor permissionExecutor;
+                try {
+                    // Try to find a constructor using a request scope first
+                    Constructor<? extends PermissionExecutor> ctor =
```

Comment:
I dont' think you need to make one of these per request.  The security test framework can just instantiate different instances of Elide with different PermissionExecutors.  I would remove this construction and just attach the executor in the builder.